### PR TITLE
feat: added support for symbol overrides

### PR DESCRIPTION
**Pull request by diana-wang**
Issue #60 

_Description of changes:_
- updated createInstance to take in an optional list of overrides
- added updateInstance function to update overrides for existing Symbol Instances

Nested Symbol overrides are currently not supported

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: models/Layer/Layer.js [hunk 1]
```diff
@@ -99,6 +99,35 @@ class Layer {
...
+          list.push(layer);
+          getRecursiveLayers(layer, list);
+        });
+      }
+      return list;
+    }
+
+    const allLayers = getRecursiveLayers(this, []);
+
+    if (predicate) {
```
Comment (dbanksdesign) [inline models/Layer/Layer.js]: This probably won't be an issue now, but in the future we might want to optimize this to only do a single pass so we don't create the full array then filter the array. But I like the readability and cleanliness of this for now, so lets keep it until we get performance issues.

File: models/SymbolMaster/SymbolMaster.test.js [hunk 2]
```diff
@@ -63,4 +67,239 @@ describe('SymbolMaster', () => {
...
+      height: 100,
+      x: 0,
+      y: 0,
+      name: 'rectangle',
+      style: beforeStyle,
+    });
+
+    symbolMaster.addLayer(layer, true);
+
+    // symbolMaster.addLayer currenty sets the overrideName incorrectly - temp fix
```
Comment (dbanksdesign) [inline models/SymbolMaster/SymbolMaster.test.js]: Could we fix this? Or maybe we do this in another PR

File: models/SymbolMaster/SymbolMaster.test.js [hunk 3]
```diff
@@ -63,4 +67,239 @@ describe('SymbolMaster', () => {
...
+    });
+
+    const overrideText = 'after';
+
+    symbolMaster.addLayer(layer, true);
+    const symbolInstance = symbolMaster.createInstance({ name: 'instance' });
+
+    symbolMaster.updateInstance(symbolInstance, [{ name: 'text', value: overrideText }]);
+
+    console.log(`result: ${symbolInstance.overrideValues[0].value}`);
```
Comment (dbanksdesign) [inline models/SymbolMaster/SymbolMaster.test.js]: Could we remove this if it isn't necessary for testing?

File: models/SymbolMaster/SymbolMaster.js [hunk 4]
```diff
@@ -81,10 +81,57 @@ class SymbolMaster extends Artboard {
...
+   * Creates a new SymbolInstance with overrides
+   * Nested Symbols are not currently supported
+   * @property {Object} [args]
+   * @property {Object[]} [args.overrideValues] - overrides
+   * @property {string} [args.overrideValues[].name] - name of the override being set
+   * @property {string|Object} [args.overrideValues[].value] - the value set, for Layer Styles pass in object or do_objectID
+   * @property {string|Object} [args.overrideValues[].style] - for textStyles only, pass in TextStyle object or do_objectID
+   * @returns {SymbolInstance}
+   */
   createInstance(args) {
```
Comment (dbanksdesign) [inline models/SymbolMaster/SymbolMaster.js]: I think you might need to rebase your branch, we changed `Object.assign()` to the spread operator for class instantiation

**Comment by diana-wang** [comment]
No worries. Thanks for looking at it!

I do have a fix now for `.addLayer`. Was going to put it in a different pull request, but it's probably better if I just merge it in with this one.

File: models/SymbolMaster/SymbolMaster.test.js (hunk 3 above)
Comment (diana-wang) [reply models/SymbolMaster/SymbolMaster.test.js]: Yup. That was a mistake. Didn't mean to include it.

File: models/Layer/Layer.test.js [hunk 5]
```diff
@@ -11,12 +11,70 @@
...
+
+    innerGroup.addLayer(text);
+    outerGroup.addLayer(innerGroup);
+
+    const layers = outerGroup.getLayers().map(l => l.name);
+
+    expect(layers.sort()).toEqual(['inner group'].sort());
+  });
+
+  it('getAllLayers should get all children layers', () => {
```
[the remark this answers is not in the document]
Comment (diana-wang) [reply models/Layer/Layer.test.js]: I agree. It'd definitely make the file easier to read. 

I'll be adding a few new tests for `.addLayer`, so it may make sense to separate out the SymbolMaster tests into 'createInstance' and 'addLayer' blocks as well

File: models/SymbolMaster/SymbolMaster.js (hunk 4 above)
Comment (diana-wang) [reply models/SymbolMaster/SymbolMaster.js]: Rebased. Should be good now?

Comment (dbanksdesign) [reply models/SymbolMaster/SymbolMaster.js]: Looks good! 👍

File: models/SymbolMaster/SymbolMaster.test.js (hunk 2 above)
Comment (diana-wang) [reply models/SymbolMaster/SymbolMaster.test.js]: committed the changes for this

File: models/SymbolMaster/SymbolMaster.test.js (hunk 3 above)
Comment (diana-wang) [reply models/SymbolMaster/SymbolMaster.test.js]: removed

File: models/Layer/Layer.test.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (diana-wang) [reply models/Layer/Layer.test.js]: updated/reorganized tests in both files
